Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -272,6 +272,66 @@ public override void Disable()
 
         #endregion IMixedRealityService Implementation
 
+        #region GetDataProvider(s) Implementation
+        /// <inheritdoc />
+        public IReadOnlyList<IMixedRealityDataProvider> GetDataProviders()
+        {
+            return new List<IMixedRealityInputDeviceManager>(deviceManagers) as IReadOnlyList<IMixedRealityInputDeviceManager>;
+        }
+
+        /// <inheritdoc />
+        public IReadOnlyList<T> GetDataProviders<T>() where T : IMixedRealityDataProvider
+        {
+            if (!(typeof(T) is IMixedRealityInputDeviceManager)) { return null; }
```

Comment:
>if (!(typeof(T) is IMixedRealityInputDeviceManager)) { return null; } [](start = 12, length = 69)

I think that here and below we want to be checking for subtypes as well - i.e. it should be something that is an implementor of device manager. So for example, I would expect to be able to ask for IMixedRealityEyeGazeDataProvider, but I can't with this code.

I tried this out locally (i.e. just having a cube that has the PanZoomBase.cs code) and without this subtype check it will basically end up returning null all the time.

Reply:
Same comment for the GetDataProvider overload down below.

---
In reply to: [285707829](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4466#discussion_r285707829) [](ancestors = 285707829)

Reply:
There were a few things going on here:
1. The in-editor simulation wasn't properly registering itself as an `IMixedRealityEyeGazeDataProvider`
2. You're right that this check wasn't sufficient to catch those cases anyway

I've updated both.